backbone.js

```diff
@@ -1993,10 +1994,10 @@
       // Normalize the fragment.
       fragment = this.getFragment(fragment || '');
 
-      // Don't include a trailing slash on the root.
+      // Strip trailing slash on the root unless _trailingSlash is true
       var rootPath = this.root;
-      if (fragment === '' || fragment.charAt(0) === '?') {
-        rootPath = rootPath.slice(0, -1) || '/';
+      if (!this._trailingSlash && (fragment === '' || fragment.charAt(0) === '?')) {
+        rootPath = rootPath.slice(0, -1) || '/'; // rootPath must always have at least '/'
```

Comment:
If you feel this comment is necessary, please put it on a separate line above. I think it doesn't add much here, so you can also leave it out completely.